cloud/shared/bin/lib/health.sh

```diff
@@ -18,7 +18,7 @@ function health::wait_for_success() {
   echo "Polling ${1} for successful response. This may take a few minutes"
   local health_status="$(health::get_status "${1}")"
 
-  until [[ "${health_status}" -eq "200" ]]; do
+  until [[ "${health_status}" -eq "200" ]] || [[ "${health_status}" -eq "303" ]]; do
```

Comment:
So ${AZURE_PRIMARY_URL}/programs (and `${AZURE_PRIMARY_URL}`) when the app is up and running returns 303 because the initial http response is a redirect to /callback?client_name=GuestClient 

I could curl /callback?client_name=GuestClient directly which would return a 302 but it doesn't seem a whole lot better. 

I'm not quite sure why the initial requset is always a redirect code but I've found that it's a reliable indicator that the app service is up and working, which is what I think this health check is supposed to cover.

Reply:
Thanks for explaining - it looks like if you don't have a session it always redirects you to the callback to establish a guest one (I can repro in an incognito window). My concern with the change is that it's a shared script and the name "wait_for_success" sounds like it would wait for 200 rather than 303. Looks like it's not used anywhere else, though, and the 303 behavior is not Azure-specific, so I guess it's okay. Maybe add a comment as to why 303 is expected though? And maybe change the name of the function to something like wait_for_healthy since 303 is not success, strictly speaking?

Reply:
Here's where that redirect happens: https://github.com/civiform/civiform/blob/a0a95e4ff507081253ddc085e6c9c630df1872cb/server/app/auth/DefaultToGuestRedirector.java

Reply:
sg!